arch/risc-v/src/mpfs/mpfs_corespi.c

```diff
@@ -0,0 +1,1589 @@
+ *         are data.
+ *
+ * Returned Value:
+ *   OK unless an error occurs.  Then a negated errno value is returned
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SPI_CMDDATA
+static int mpfs_spi_cmddata(struct spi_dev_s *dev,
+                            uint32_t devid, bool cmd)
+{
+  spierr("SPI cmddata not supported\n");
+  DEBUGPANIC();
+
+  return -1;
+}
+#endif
+
+/****************************************************************************
+ * Name: mpfs_spi_hwfeatures
+ *
+ * Description:
+ *   Set hardware-specific feature flags.
+ *
+ * Input Parameters:
+ *   dev      - Device-specific state data
+ *   features - H/W feature flags
+ *
+ * Returned Value:
+ *   Zero (OK) if the selected H/W features are enabled; A negated errno
+ *   value if any H/W feature is not supportable.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SPI_HWFEATURES
+static int mpfs_spi_hwfeatures(struct spi_dev_s *dev,
+                               spi_hwfeatures_t features)
+{
+  /* Other H/W features are not supported */
+
+  spierr("SPI hardware specific feature not supported\n");
+  DEBUGPANIC();
+
+  return -1;
+}
+#endif
+
+/****************************************************************************
+ * Name: mpfs_spi_sem_waitdone
+ *
+ * Description:
+ *   Wait for a transfer to complete.
+ *
+ * Parameters:
+ *   priv          - Pointer to the internal driver state structure.
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success. A negated errno value is returned on
+ *   failure.
+ *
+ ****************************************************************************/
+
+static int mpfs_spi_sem_waitdone(struct mpfs_spi_priv_s *priv)
+{
+  uint32_t timeout = SPI_TTOA_US(priv->txwords * priv->nbits, priv->actual);
+  timeout += SPI_TTOA_MARGIN_US;
+  return nxsem_tickwait_uninterruptible(&priv->sem_isr, USEC2TICK(timeout));
+}
+
+/****************************************************************************
+ * Name: mpfs_spi_load_tx_fifo
+ *
+ * Description:
+ *   Fill up the TX fifo with nwords (in 8 or 16-bit).
+ *
+ * Input Parameters:
+ *   priv     - SPI private state data
+ *   txbuffer - A pointer to the buffer of data to be sent
+ *   nwords   - the length of data that to be exchanged in units of words.
+ *              The wordsize is determined by the number of bits-per-word
+ *              selected for the SPI interface.  If nbits <= 8, the data is
+ *              packed into uint8_t's; if nbits >8, the data is packed into
+ *              uint16_t's
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void mpfs_spi_load_tx_fifo(struct mpfs_spi_priv_s *priv,
+                                  const void *txbuffer,
+                                  uint32_t nwords)
+{
+  uint16_t *data16;
+  uint8_t  *data8;
```

Comment:
```suggestion
  uint8_t *data8;
```

Reply:
Yes. Sure. Just either remove space or apply alignment to `last` and `i` below.

Reply:
Seems like the style you suggest is used more consistently in the file (this is what happens when you take an existing driver as a base and modify it, it is called a mess..)